qiita_db/support_files/patches/python_patches/53.py

```diff
@@ -0,0 +1,19 @@
+from qiita_db.study import Study
+
+studies = Study.get_by_status('private').union(
+    Study.get_by_status('public')).union(Study.get_by_status('sandbox'))
+raw_data = [pt.artifact for s in studies for pt in s.prep_templates()]
+
+for rd in raw_data:
+    # getting the most open visibility of all the children in the pipeline
+    children = rd.descendants.nodes()
+    vis = [a.visibility for a in children]
+    vis.append(rd.visibility)
+
+    new_vis = 'sandbox'
+    if 'public' in vis:
+        new_vis = 'public'
+    elif 'private' in vis:
```

Comment:
are there situations where an artifact was bumped to private but where the user expects all other artifacts to remain sandboxed?